xbmc/cores/AudioEngine/Sinks/AESinkDARWINOSX.cpp

```diff
@@ -93,13 +93,56 @@ static CADeviceList GetDevices()
   return list;
 }
 
+//---------- DEVICE CHANGED EVENT HANDLING
+// when device list change callbacks are received
+// inform the ae engine this amount of time after
+// the last callback was received (bundle multiple callbacks)
+// into one final call
+#define DEVICE_LIST_CHANGED_TIMER_MS 1500
+
+CDeviceChangeListTimeout::CDeviceChangeListTimeout()
+{
+  m_pDeviceListChangedTimer = new CTimer(this);
```

Comment:
The latter case would just construct a new sink, with a new member on stack, initialized to expired. Yes - you are right with the other arguments. Though, with a little tweaking it can be extended to also match this usecase (another member during AddPackets (m_deviceChanged = true after first event + Set(1500) for every consequent event), if m_deviceChanged and Timer elapsed -> Rescan.

If the sink is deleted while a timer is running, the timer should (!) go away, cause AE is reiniting itself already and will create new sinks anyways. The timer is not aware of this situation, it would just falsely trigger AE again, but AE just did a new enumeration by itself and is uptodate.

Whatever you prefer.